src/projectviewer/projectviewermodel/recentprojectsmodel.cpp

```diff
@@ -0,0 +1,32 @@
+#include "recentprojectsmodel.h"
+
+#include <QDebug>
+#include <QIcon>
+
+RecentProjectsModel::RecentProjectsModel(const ProgramSettings &settings, QObject *parent)
+    :mSettings{settings}
+{
+
+}
+
+int RecentProjectsModel::rowCount(const QModelIndex &parent) const
+{
+    return mSettings.recentProjects().size();
+}
+
+
+QVariant RecentProjectsModel::data(const QModelIndex &index, int role) const
+{
+    int row = index.row();
```

Comment:
At the beginning of the `data` method, you should check whether `index` is valid, if not return empty QVariant. Also, if you create a model, QtCreator has a wizard for model creation and it will add this code with checks for you.